jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java

```diff
@@ -118,4 +121,108 @@ public String describeQueryResultColumns(final String query) throws SQLException
+                    "SQL Dialect \"" + this.properties.getSqlDialect() + "\" is not supported. " + availableDialects);
+        }
+        if (this.properties.containsKey(CONNECTION_NAME_PROPERTY)) {
+            if (this.properties.containsKey(CONNECTION_STRING_PROPERTY)
+                    || this.properties.containsKey(USERNAME_PROPERTY)
+                    || this.properties.containsKey(PASSWORD_PROPERTY)) {
+                throw new PropertyValidationException("You specified a connection (" + CONNECTION_NAME_PROPERTY
+                        + ") and therefore may not specify the properties " + CONNECTION_STRING_PROPERTY + ", "
+                        + USERNAME_PROPERTY + " and " + PASSWORD_PROPERTY);
+            }
+        } else {
+            if (!this.properties.containsKey(CONNECTION_STRING_PROPERTY)) {
+                throw new PropertyValidationException("You did not specify a connection (" + CONNECTION_NAME_PROPERTY
+                        + ") and therefore have to specify the property " + CONNECTION_STRING_PROPERTY);
+            }
+        }
+    }
+
+    protected void validateBooleanProperty(final String property) throws PropertyValidationException {
+        if (this.properties.containsKey(property) //
+                && !this.properties.get(property).toUpperCase().matches("^TRUE$|^FALSE$")) {
+            throw new PropertyValidationException("The value '" + this.properties.get(property) + "' for the property "
+                    + property + " is invalid. It has to be either 'true' or 'false' (case insensitive).");
+        }
+    }
+
+    private void validateDebugOutputAddress(final String debugAddress) throws PropertyValidationException {
+        if (!debugAddress.isEmpty()) {
+            final String error = "You specified an invalid hostname and port for the udf debug service ("
+                    + DEBUG_ADDRESS_PROPERTY + "). Please provide a valid value, e.g. 'hostname:3000'";
+            if (debugAddress.split(":").length != 2) {
+                throw new PropertyValidationException(error);
+            }
+            try {
+                Integer.parseInt(debugAddress.split(":")[1]);
+            } catch (final NumberFormatException ex) {
+                throw new PropertyValidationException(error);
+            }
+        }
+    }
+
+    private void validateExceptionHandling(final String exceptionHandling) throws PropertyValidationException {
+        if (!((exceptionHandling == null) || exceptionHandling.isEmpty())) {
+            for (final AbstractSqlDialect.ExceptionHandlingMode mode : AbstractSqlDialect.ExceptionHandlingMode
+                    .values()) {
+                if (!mode.name().equals(exceptionHandling)) {
+                    throw new PropertyValidationException(
+                            "You specified an invalid exception mode (" + exceptionHandling + ").");
+                }
+            }
+        }
+    }
+
+    protected String getProperty(final String name) {
```

Comment:
Check `isEnabled()` in `AdapterProperties` and get rid of the `getProperty(...)` methods here.

Reply:
fixed